hawk_eye/data_generation/slice_image.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+""" Contains code to slice up an image into smaller tiles. """
+
+import argparse
+import pathlib
+import pdb
+import sys
+from typing import Tuple
+
+import numpy as np
+from PIL import Image
+
+from hawk_eye.data_generation import generate_config as config
+
+
+def slice_image(
+    img_dir: str, tile_size: Tuple[int, int], overlap: int, save_dir: str
+) -> None:
+    """ Take in an image and slice it into smaller images.
+    Args:
+        img_dir: The path to the directory with images to be sliced.
+        tile_size: The (width, height) of the tiles.
+        overlap: The overlap between adjacent tiles.
+        save_dir: The path to the directory within which to save slices.
+    Returns:
+        None.
+    Usage:
+        >>> slice_image('hawk-eye/hawk_eye/data_generation/data/test_flight_targets_20190215', (512, 512), 50, 'hawk_eye/data_generation/data/slices')
```

Comment:
See how doctests are added to the unittest files [here](https://github.com/uavaustin/hawk-eye/blob/master/hawk_eye/test/test_third_party.py). Add a similar setup to `hawk_eye/test/test_data_generation.py`

Reply:
I haven't implemented testing this function yet